### PR TITLE
feat: update message for enrolment, renewal, error modal when grace period is over

### PR DESCRIPTION
**Pull request by arjita-mitra**
## Description
update message for enrolment, renewal, error modal when grace period is over

## Screenshots/Screencast (for UI changes)
Screenshot 2024-01-18 at 16 08 55
Screenshot 2024-01-18 at 16 09 00
Screenshot 2024-01-18 at 16 09 16
Screenshot 2024-01-18 at 16 09 21

Screenshot 2024-01-18 at 16 09 28

- [x] PR has been self reviewed by the author;
- [] Hard-to-understand areas of the code have been commented;
- [ ] If it is a core feature, unit tests have been added;

### CODE REVIEW

File: src/script/E2EIdentity/E2EIdentityEnrollment.ts [hunk 1]
```diff
@@ -365,13 +365,13 @@ export class E2EIHandler extends TypedEventEmitter<Events> {
     // Clear the e2e identity progress
     this.coreE2EIService.clearAllProgress();
 
-    const isSoftLockEnabled = await shouldEnableSoftLock(this.config!);
+    const disableSnooze = await shouldEnableSoftLock(this.config!);
```
Comment (atomrc) [inline src/script/E2EIdentity/E2EIdentityEnrollment.ts:368]: 👍  nice renaming

File: src/script/E2EIdentity/Modals/Modals.ts [hunk 2]
```diff
@@ -33,14 +33,28 @@ export enum ModalType {
...
+ * @property {Function} [primaryActionFn] - Function to execute for the primary action.
+ * @property {Function} [secondaryActionFn] - Function to execute for the secondary action.
+ * @property {boolean} [hideSecondary=false] - Flag to hide the secondary action button.
+ * @property {boolean} [hidePrimary=false] - Flag to hide the primary action button.
+ * @property {boolean} [hideClose=false] - Flag to hide the close button.
+ * @property {Object} [extraParams] - Additional parameters for the modal.
+ * @property {string} [extraParams.delayTime] - time left to remind the user again for enroll/renewal.
+ * @property {boolean} [extraParams.isRenewal] - Flag indicating if this is a renewal action.
+ * @property {boolean} [extraParams.isGracePeriodOver] - Flag indicating if the grace period is over.
+ */
```
Comment (atomrc) [inline src/script/E2EIdentity/Modals/Modals.ts]: Oh, you did it the hard way. There is actually a super nice jsdoc/typescript feature where you can inline jsdoc directly in the types. I'll show you in a diff comment

File: src/script/E2EIdentity/Modals/Modals.ts [hunk 3]
```diff
@@ -33,14 +33,28 @@ export enum ModalType {
...
 interface GetModalOptions {
   type: ModalType;
   primaryActionFn?: () => void;
   secondaryActionFn?: () => void;
   hideSecondary?: boolean;
   hidePrimary?: boolean;
   hideClose?: boolean;
-  extraParams?: {delayTime?: string; isRenewal?: boolean};
+  extraParams?: {delayTime?: string; isRenewal?: boolean; isGracePeriodOver?: boolean};
```
Comment (atomrc) [inline src/script/E2EIdentity/Modals/Modals.ts]:
```suggestion
  extraParams?: {
    /** time left to remind the user again (only for enroll and renewal modal types). */
    delayTime?: string; 
    
    /** Flag indicating if this is a renewal action (only for success and loading modals) */
    isRenewal?: boolean; 
    
    /** Flag indicating if the grace period is over (only for enroll, renew or error modals) */
    isGracePeriodOver?: boolean};
```

**Review by atomrc — APPROVED** [review]

File: src/script/E2EIdentity/Modals/Modals.ts (hunk 2 above)
Comment (arjita-mitra) [reply src/script/E2EIdentity/Modals/Modals.ts]: Ahh ok, shall i remove my jsdoc then?

Comment (atomrc) [reply src/script/E2EIdentity/Modals/Modals.ts]: I think you can remove them then :) 
Inline jsdoc has the benefit of showing up on the typescript server language :)

**Pull request merged**